File: samples/java/src/main/java/co/windly/limbosample/application/ApplicationComponent.java
package co.windly.limbosample.application;

import co.windly.limbosample.presentation.PresentationModule;
import dagger.Component;
import javax.inject.Singleton;

@Singleton
@Component(modules = { PresentationModule.class })
interface ApplicationComponent {

  //region Injection

  void inject(LimboSample application);

  //endregion

  //region Component Builder

  @Component.Builder
  interface Builer {

    ApplicationComponent build();
  }

  //endregion

  //region Component Provider

  interface ComponentProvider {

    ApplicationComponent getApplicationComponent();
  }

  //endregion
}

File: examples/mvp-kotlin/src/main/java/co/windly/limbosample/application/ApplicationComponent.kt
package co.windly.limbosample.application

import co.windly.limbosample.presentation.PresentationModule
import dagger.Component
import javax.inject.Singleton

@Singleton
@Component(modules = [PresentationModule::class])
interface ApplicationComponent {

  //region Injection

  fun inject(application: LimboSample)

  //endregion

  //region Component Builder

  @Component.Builder
  interface Builder {

    fun build(): ApplicationComponent
  }

  //endregion

  //region Component Provider

  interface ComponentProvider {

    val applicationComponent: ApplicationComponent
  }

  //endregion
}
